### ISSUE TITLE
Idea to add custom validator for IP address vs prefix

### ISSUE THREAD

**Issue by tzarski0** [netascode/nac-validate#280]
We are using the IPv4/IPv6 addresses and prefixes a lot. The build-in YAMALE ip validator (for example, ip(version=4)) accepts both address (for example 10.0.0.1) and prefix (10.0.0.0/24). However in most cases we want to validate either address or prefix. The idea is to create custom validator for ip_address and ip_prefix so for example:
- in prefix-list we can validate the prefix (e.g. 10.0.0.0/24 has both address and / subnet mask)
- in logging/tacacs server ip we can validate the address (e.g. 10.0.0.0 has IP, but doesn't have subnet mask).

**Comment by kuba-mazurkiewicz** [comment]
Hey @tzarski0 ,

for validating IP address with or without prefix you can use regex instead of built in yamale ip validator

For Ipv4 without prefix you can use something like that:

```^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}$```

for IPv4 with prefix:

```^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}\/(3[0-2]|[1-2]?[0-9])$```

And for ipv6 without prefix:

```^([0-9a-fA-F]{1,4}:){7}([0-9a-fA-F]{1,4})$```

Ipv6 with prefix:

```^([0-9a-fA-F]{1,4}:){7}([0-9a-fA-F]{1,4})\/(12[0-8]|1[01][0-9]|[1-9]?[0-9])$```

**Comment by tzarski0** [comment]
True, it's just that then the error that iac-validate shows is "... is not a valid regex: ^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}$", which is not very readable for end user. We could probably make it more clear if we implemented a simple ip address and prefix validator.
